Fix reversing y-axis of diffraction patterns in template matching

rbjorge opened the pull request:
**What does this PR do?**
This pull request seeks to fix the problem raised in issue #925. In short, diffraction patterns to be template matched currently have the y-axis reversed when template matching is performed (when they are converted to polar coordinates). This PR modifies `_warp_polar_custom()` to make the polar angle increase counter-clockwise when converting to polar coordinates. This also required changing `plot_template_over_pattern()`.

The root cause of this issue, as far as I can see, are the two different coordinate system origin conventions used: top-left origin in hyperspy/pyxem and bottom-left in diffsims. I think both of these make sense in their respective contexts.

There are three failing test functions:
- [`tests.utils.test_indexation_utils.test_results_dict_to_crystal_map()`](https://github.com/pyxem/pyxem/blob/a0476c6cc2943650ce46230ce12958e8396af985/pyxem/tests/utils/test_indexation_utils.py#L73)
-  [`tests.utils.test_rotation_tempate_match.test_match_image_to_template()`](https://github.com/pyxem/pyxem/blob/a0476c6cc2943650ce46230ce12958e8396af985/pyxem/tests/utils/test_rotation_tempate_match.py#L91)
-  [`tests.utils.test_rotation_tempate_match.test_match_image_to_template_gpu()`](https://github.com/pyxem/pyxem/blob/a0476c6cc2943650ce46230ce12958e8396af985/pyxem/tests/utils/test_rotation_tempate_match.py#L122)

It's not surprising that these tests fail, so I think it is best that I get some initial feedback on this PR before I spend time modifying the tests.

**To do**
- [x] Fix failing tests
- [x] Document conventions
- [x] List in changelog

hakonanes commented:
Thanks for taking the time to troubleshoot this inconsistency, @rbjorge! I hope @din14970 finds time to comment.

Things left to do in this PR, as far as I can see:
- Update tests
- List this (breaking) fix in the changelog
- Update any affected tutorials

CSSFrancis commented:
@rbjorge I don't see a reason why this can't be merged once we clean up the tests, but I'd like @din14970 to comment on it first!

Out of curiosity does this match the frame of reference for Kikuchipy?  I think we should try to be as internally consistent as possible. 

Let me know if you need any help adding tests, or fixing them!

rbjorge commented:
I've now fixed the tests and added the notebook on conventions (not sure where to put the png file, so I put it in the same folder as the notebooks).

I don't think that the notebooks on [readthedocs.io](https://pyxem.readthedocs.io/en/stable/) are affected by this PR (I don't see any template matching there). However, some of the notebooks in pyxem/pyxem-demos are affected. Mainly, since the diffraction patterns are no longer flipped, the value of `mirrored_template` will usually switch, as expected (same for `signs` in `get_n_best_matches()`). However, because of the "noisy" results of template matching, sometimes a different solution will be picked instead. When I checked this PR on a dataset with 192,000 experimental patterns, the value of `mirrored_template` was switched for 97-98 % of the patterns.

For the notebook [`11 Accelerated orientation mapping with template matching.ipynb`](https://github.com/pyxem/pyxem-demos/blob/master/11%20Accelerated%20orientation%20mapping%20with%20template%20matching.ipynb) this means that it is now the mirrored correlations that are the highest. There are some small changes to the IPF maps, but when overlaying the best-matching template on the experimental pattern it still looks reasonable.

Also, since the value of the Euler angles are changed by this PR, user scripts that rely on the actual numerical values will be affected (for example [this one](https://github.com/pyxem/orix/issues/452) perhaps).

Maybe this should somehow be included in the changelog entry.

rbjorge commented:
Thanks for getting this merged, @CSSFrancis . I see now that one consequence of this fix is that datasets that were template matched before this change, will have the template flipped when using `plot_template_over_pattern()` with this fix.
